### PR TITLE
[RFR] Add support for CFME QCOW2 templates for RHV

### PR DESCRIPTION
**Pull request by nachandr**
Added support for upload CFME QCOW2 templates for RHV.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 926725ef8ba6e544d6827e05a16c268e66725177:

`scripts/template_upload_all.py`:
- :red_circle: Line 19:1: **F401** *'pdb' imported but unused*

`scripts/template_upload_rhevm_qcow2.py`:
- :red_circle: Line 11:1: **F401** *'pdb' imported but unused*
- :red_circle: Line 15:1: **F401** *'re' imported but unused*
- :red_circle: Line 21:1: **F401** *'cfme.utils.trackerbot' imported but unused*
- :red_circle: Line 343:18: **F821** *undefined name 'check_edomain_template'*
- :red_circle: Line 351:18: **F821** *undefined name 'is_edomain_template_deleted'*
- :red_circle: Line 528:13: **F821** *undefined name 'import_template_glance'*

Hmmm
:frowning: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d74472501cdf5b0b3cc54afd07a6c9df295c5fc1:

`scripts/template_upload_all.py`:
- :red_circle: Line 19:1: **F401** *'pdb' imported but unused*

`scripts/template_upload_rhevm_qcow2.py`:
- :red_circle: Line 11:1: **F401** *'pdb' imported but unused*
- :red_circle: Line 15:1: **F401** *'re' imported but unused*
- :red_circle: Line 21:1: **F401** *'cfme.utils.trackerbot' imported but unused*
- :red_circle: Line 158:1: **F831** *duplicate argument 'provider' in function definition*

Hmmm
:frowning: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

File: scripts/image_upload_glance.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def upload_to_glance(image, image_name_in_glance, provider, disk_format):
     glance.images.upload(glance_img.id, open(image, 'rb'))
 
 
-def main():
+def main(args):
```
Comment (mshriver) [inline scripts/image_upload_glance.py]: So I think I'd like to see this re-arranged to function a bit cleaner.

1. rename this method to `run`
2. Move argparse into separate method `parse_args` (something like that)
3. `if __name__ == '__main__'` block modified to call `parse_args` and pass it to `run`

I think doing it that way is cleaner, and allows for easy calling from within a module or on the command line.

File: scripts/template_upload_all.py [hunk 2]
```diff
@@ -255,7 +256,7 @@ def browse_directory(dir_url):
         logger.exception("Skipping: %r", dir_url)
         return None
 
-    rhevm_pattern = re.compile(r'<a href="?\'?([^"\']*(?:rhevm\.ova|ovirt)[^"\'>]*)')
+    rhevm_pattern = re.compile(r'<a href="?\'?([^"\']*rhevm[^"\'>]*)')
```
Comment (mshriver) [inline scripts/template_upload_all.py]: Let's use an explicit regex here to get the qcow2 image, and not rely on the order in which these are matched.

L 275 below just picks the first image name in the list, which _could_ be the ova image instead.

File: scripts/template_upload_all.py [hunk 3]
```diff
@@ -58,6 +57,8 @@ def parse_cmd_line():
                         help='local yaml file path, to use local provider_data & not conf/cfme_data'
                              'to be useful for template upload/deploy by non cfmeqe',
                         default=None)
+    parser.add_argument("--glance", dest="glance",
+                        help="Glance server to upload images to", default='glance11-server')
```
Comment (mshriver) [inline scripts/template_upload_all.py]: Would like to see this server in yamls before merging.

File: scripts/template_upload_all.py [hunk 4]
```diff
@@ -390,6 +392,7 @@ def main():
             return 1
         kwargs['stream'] = stream
         kwargs['image_url'] = dir_files[module]
+        kwargs['glance'] = glance
```
Comment (mshriver) [inline scripts/template_upload_all.py]: This is speaking in general, not just to this line.

I'm thinking we should hold off on modifying template_upload_all for the moment and leave it still using the existing ova upload script.

We can run this script manually and do some more testing with sprout before switching over to using glance/qcow2. I think we'd be rushing to modify it in this sprint.

**Comment by nachandr** [comment]
@mshriver , I've undone all the changes to template_upload_all.py so that it can use the existing ova upload script.

This PR requires that https://gitlab.cee.redhat.com/cfme-qe/cfme-qe-yamls/merge_requests/366 .
PR # 366 adds CFME Template Hardware configuration details.

File: scripts/image_upload_glance.py (hunk 1 above)
Comment (nachandr) [reply scripts/image_upload_glance.py]: @mshriver , done

File: scripts/image_upload_glance.py [hunk 5]
```diff
@@ -44,8 +57,8 @@ def upload_to_glance(image, image_name_in_glance, provider, disk_format):
     # So, we are running a check to make sure an image with the same name doesn't already exist.
     for img in glance.images.list():
         if img.name == image_name_in_glance:
-            print("Image already exists on Glance server")
-            sys.exit(127)
+            print("image_upload_glance: Image already exists on Glance server")
```
Comment (mshriver) [inline scripts/image_upload_glance.py]: **Optional**
You can use `__file__.rstrip('.py')` here.

**Pull request merged**